sonic_platform_base/sfp_base.py

```diff
@@ -0,0 +1,204 @@
+        Retrieves the tx-disable status of this SFP
+
+        Returns:
+            A Boolean, True if tx-disable is enabled, False if disabled
+        """
+        return NotImplementedError
+
+    def get_tx_disable_channel(self):
+        """
+        Retrieves the tx disabled channels in this SFP
+
+        Returns:
+            A hex of 4 bits (bit 0 to bit 3 as channel 0 to channel 3) to represent
+            tx channels which have been disabled in this SFP.
+            As an example, a returned value of 0x5 indicates that channel 0 
+            and channel 2 have been disabled.
+        """
+        return NotImplementedError
+
+    def get_lpmode(self):
+        """
+        Retrieves the lpmode status of this SFP
+
+        Returns:
+            A Boolean, True if lpmode (low power mode) is enabled, False if disabled
+        """
+        return NotImplementedError
+
+    def get_power_override(self):
+        """
+        Retrieves the power-override status of this SFP
+
+        Returns:
+            A Boolean, True if power-override is enabled, False if disabled
+        """
+        return NotImplementedError
+
+    def reset(self):
+        """
+        Reset SFP and return all user module settings to their default srate.
+
+        Returns:
+            A boolean, True if successful, False if not
+        """
+        return NotImplementedError
+    
+    def tx_disable(self, tx_disable):
+        """
+        Disable SFP tx for all channels
+
+        Args:
+            tx_disable : A Boolean, True to enable tx_disable mode, False to disable
+                         tx_disable mode.
+
+        Returns:
+            A boolean, True if tx_disable is set successfully, False if not
+        """
+        return NotImplementedError
+    
+    def tx_disable_channel(self, channel, disable):
+        """
+        Sets the tx_disable for specified SFP channels
+
+        Args:
+            channel : A hex of 4 bits (bit 0 to bit 3) which represent channel 0 to 3,
+                      e.g. 0x5 for channel 0 and channel 2.
+            disable : A boolean, Trure to disable tx channels specified in channel,
+                      False to enable
+
+        Returns:
+            A boolean, True if successful, False if not
+        """
+        return NotImplementedError
+    
+    def set_lpmode(self, lpmode):
+        """
+        Sets the lpmode of SFP
+
+        Args:
+            lpmode: A Boolean, True to enable lpmode (low power mode),
+                    False to disable it
+            Note  : lpmode can be overridden by set_power_override
+
+        Returns:
+            A boolean, True if lpmode is set successfully, False if not
+        """
+        return NotImplementedError
+    
+    def set_power_override(self, power_override, power_set):
+        """
+        Sets SFP power level using power_override and power_set
+
+        Args:
+            power_override : A Boolean, True to override lpmode and have power_set
```

Comment:
This description is a bit confusing. Can you make it more clear?

Reply:
Changed the descriptions to 
        Args:
            power_override :
                    A Boolean, True to override set_lpmode and have power_set
                    to control SFP power, False to disable SFP power control
                    through power_override/power_set and have set_lpmode
                    to control SFP power.
            power_set :
                    Only valid when power_override is True.
                    A Boolean, True to set power at low power mode, False to set
                    SFP to high power mode.